apps/src/applab/applab.js

```diff
@@ -1012,6 +1013,7 @@ Applab.reset = function () {
     designMode.resetPropertyTab();
   }
 
+  getStore().dispatch(disconnectMaker());
   if (makerBoard) {
     makerBoard.destroy();
     makerBoard = null;
```

Comment:
I'll happily dispatch `disconnectMaker()` after `makerBoard.destroy()` but I think you've identified a deeper issue here.  I don't believe `board.destroy()` does cancel anything in `board.connect()` (a limitation of promise chains - they're hard to interrupt), and even if that is solved this could be a problem if there's an error while creating the board (say, finding a port) so we don't want to call `destroy()` at all.  Hmm...